app/models/invoice.rb

```diff
@@ -99,14 +99,13 @@ class Invoice < ApplicationRecord
   scope :ready_to_be_refreshed, -> { where(ready_to_be_refreshed: true) }
   scope :ready_to_be_finalized,
     lambda {
-      date = <<-SQL
-            (
-              invoices.created_at +
-              COALESCE(customers.invoice_grace_period, organizations.invoice_grace_period) * INTERVAL '1 DAY'
-            )
-      SQL
-
-      draft.joins(:customer, :organization).where("#{Arel.sql(date)} < ?", Time.current)
+      draft
```

Comment:
before we haven't used customer timezone in this scope

Reply:
issuing date is defined like this in the `/app/services/invoices/create_generating_service.rb`

```ruby
    def issuing_date
      date = datetime.in_time_zone(customer.applicable_timezone).to_date
      return date if !grace_period? || charge_in_advance

      date + customer.applicable_invoice_grace_period.days
    end
```

it means that it is technically already a date expressed in the customer timezone.
Maybe I'm wrong, but I don't think we need to do the timezone shift again.

That said, maybe a comment would help future us reading this scope and asking themselves about it :smile:

Reply:
Actually you are right! Isuuing date is already converted to customer timezone and this part should not be converted!
However I also wanted initially to convert server time in customer timezone so that we include certain invoice for finalization in correct moment. However, after discussion with Mike we concluded that we should rather not change current behaviour and I am not using any timezone shift now. E.g. server time is shifted also in job that terminates invoices based on ending_at and I wanted to follow same pattern here.